azure_functions_worker/dispatcher.py

```diff
@@ -700,6 +745,49 @@ def gen(resp_queue):
             error_logger.exception('unhandled error in gRPC thread')
             raise
 
+    def _process_indexed_function(self, indexed_functions: List[Function]):
+        fx_metadata_results = []
+        for indexed_function in indexed_functions:
+            function_id = str(uuid.uuid4())
+            function_info = self._functions.add_indexed_function(
+                function_id,
+                function=indexed_function)
+
+            binding_protos = {}
+            for binding in indexed_function.get_bindings():
+                binding_protos[binding.name] = protos.BindingInfo(
+                    type=binding.type,
+                    data_type=binding.data_type,
+                    direction=binding.direction)
+
```

Comment:
would be good to extract this into a function like build_binding_protos(indexed_function)